networking_nsxv3/common/synchronization.py

```diff
@@ -142,33 +142,30 @@ def run(self, priority, ids, fn):
         """
         for jid in ids:
             try:
-                LOG.info(MESSAGE.format(fn.__name__,jid, priority, "enqueued"))
+                LOG.info(MESSAGE.format("Enqueued", jid, priority.name, fn.__name__))
 
                 job = Runnable(jid, fn, priority.value)
-
-                job = (priority.value, {"id": jid, "fn": fn})
                 if priority.value == Priority.HIGHEST:
                     self._active.put_nowait(job)
                 else:
                     self._passive.put_nowait(job)
             except eventlet.queue.Full as err:
-                LOG.error(MESSAGE.format(fn.__name__, jid, priority, err))
+                LOG.error(MESSAGE.format(err, jid, priority.name, fn.__name__))
 
     def _start(self):
         while True:
```

Comment:
I would expect that there is some way to actually stop the loop.

Reply:
That would be fine, but personally, I would prefer an explicit variable, e.g. something like `_terminating`. But is that sufficient? I mean, isn't there work done in the loop which needs to be finished before terminating? It's an honest question, I haven't understood the code yet to that extend.

Reply:
The pool size can get to zero only if a termination operation is triggered. According to the documentation all jobs needs to finish prior the size is set to 0, thus it will be safe to use this as a termination trigger.